### PR TITLE
Feat(server)/implement youtube api V1

### PR DESCRIPTION
**Pull request by khoaquocngo**
#17

### CODE REVIEW

File: apps/server/src/datasources/utils/controllers/youtube/searchVideo.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import findVal from './helpers/findVal';
+import formatVideo from './helpers/formatVideo';
+import getData from './helpers/getData';
+import Video from './types/video';
+
+export default async function searchVideo(terms: string, token?: string, apikey?: string) {
+  try {
+    let items = [];
```
[the remark this answers is not in the document]
Comment (khoaquocngo) [reply apps/server/src/datasources/utils/controllers/youtube/searchVideo.ts:8]: can't fix!

File: apps/server/src/datasources/utils/controllers/youtube/helpers/formatVideo.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
+/* eslint-disable no-prototype-builtins */
+import getVideoDate from '../getVideoDate';
+import getDateFromText from './getDateFromText';
+
+export default async function formatVideo(video, speedDate = false) {
```
[the remark this answers is not in the document]
Comment (khoaquocngo) [reply apps/server/src/datasources/utils/controllers/youtube/helpers/formatVideo.ts:5]: can't fix!

File: apps/server/src/datasources/utils/controllers/youtube/helpers/getData.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
+/* eslint-disable no-useless-escape */
+import axios from 'axios';
+
+import decodeHex from './decodeHex';
+import findVal from './findVal';
+
+export default async function getData(urlstring: string) {
```
[the remark this answers is not in the document]
Comment (khoaquocngo) [reply apps/server/src/datasources/utils/controllers/youtube/helpers/getData.ts:7]: can't fix!

**Comment by gialynguyen** [comment]
@khoaquocngo @trungpham71198 Has any update?

**Comment by khoaquocngo** [comment]
> 

No update

**Review by vtrungpd3 — APPROVED** [review]

**Review by vtrungpd3 — APPROVED** [review]

**Pull request merged**